pyccel/epyccel.py

```diff
@@ -240,8 +240,8 @@ def epyccel( python_function_or_module, **kwargs ):
                 fun_name = python_function_or_module.__name__ if fun else None
                 success  = True
             # error handling carried out after broadcast to prevent deadlocks
-            except: # pylint: disable=bare-except
-                exc_info = sys.exc_info()
+            except BaseException as e: # pylint: disable=broad-except
+                exc_info = e
                 success  = False
```

Comment:
Are we sure that we get traceback object this way? See https://docs.python.org/3/library/sys.html#sys.exc_info

Reply:
OK, apparently the traceback is available directly through the exception, as `e.__traceback__` (see https://docs.python.org/3/reference/datamodel.html#traceback-objects).

Reply:
Yes. The traceback is available. But if we do `raise sys.exc_info()` then we get a `TypeError` because `exc_info` doesn't inherit from `BaseException`

Reply:
Perfect, thanks!